src/scenario.jl

```diff
@@ -648,51 +660,48 @@ function run_model(domain::Domain, param_set::YAXArray, functional_groups::Vecto
 
     FLoops.assistant(false)
     for tstep::Int64 in 2:tf
-        survival_rate_slices = [@view survival_rate_cache[:, :, loc] for loc in 1:n_locs]
-        apply_mortality!.(functional_groups, survival_rate_slices)
-        recruitment .*= reshape(survival_rate_cache[:, 1, :], (n_groups, n_locs))
-
-        # Determine absolute area for coral model
+        # Convert covert to absolute values to use within CoralBlox model
         C_cover_t[:, :, habitable_locs] .= C_cover[tstep-1, :, :, habitable_locs] .* habitable_loc_areas′
 
-        # Constrain growth by available area
-        growth_spatial_constraint[habitable_locs] .= log2.(1 .+ relative_leftover_space(
-            dropdims(sum(C_cover[tstep-1, :, :, habitable_locs]; dims=(1, 2)), dims=(1, 2))
-        ))
+        _loc_rel_leftover_space = relative_leftover_space(
+            dropdims(sum(C_cover[tstep-1, :, :, habitable_locs], dims=(1, 2)), dims=(1, 2))
+        )
+
+        lin_ext_scale_factors::Vector{Float64} = linear_extension_scale_factors(
+            C_cover_t[:, :, habitable_locs],
+            habitable_loc_areas,
+            linear_extensions(),
+            bin_edges(),
+            max_projected_cover[habitable_locs],
+        )
```

Comment:
Move `linear_extensions()` and `bin_edges()` to outside the loop to reduce memory allocations